otx/cli/tools/build.py

```diff
@@ -57,56 +58,102 @@ def parse_args():
+):
+    """Makes configuration files (data.yaml, template.yaml, ...) in workspace directory.
+
+    Args:
+        builder (Builder): Builder class
+        train_data_roots (str, optional): The path of training dataset. Defaults to None.
+        val_data_roots (str, optional): The path of validation dataset. Defaults to None.
+        task (str, optional): The name of task (i.e. classification, detection, segmentation, ..). Defaults to None.
+        train_type (str, optional): The type of training (i.e. incremental, semi-sl, ..). Defaults to "incremental".
+        workspace_root (str, optional): The path of workspace. Defaults to None.
+        model (str, optional): The path for model.py. Defaults to None.
+        backbone (str, optional): The name of backbone models (i.e. mmdet.ResNet). Defaults to None.
+        save_backbone_to (str, optional): The path for saving model, it is used on train.py. Defaults to None.
+        otx_root (str, optional): The path for OTX root. Defaults to ".".
+        template (ModelTemplate, optional): ModelTemplate class. Defaults to None.
+
+    """
 
     # Auto-configuration
     config_manager = ConfigManager()
-    if args.train_data_roots:
-        if args.task is None:
-            task_type = config_manager.auto_task_detection(args.train_data_roots)
-            args.task = task_type
-        if args.val_data_roots is None:
-            config_manager.auto_split_data(args.train_data_roots, args.task)
-
-    if args.task and args.task in SUPPORTED_TASKS:
+    if train_data_roots:
+        if task is None:
+            task_type = config_manager.auto_task_detection(train_data_roots)
+            task = task_type
+        if val_data_roots is None:
+            config_manager.auto_split_data(train_data_roots, task)
+
+    if task and task in SUPPORTED_TASKS:
         # Build with task_type and create user workspace
-        if args.workspace_root is None:
-            args.workspace_root = set_workspace(args.task, args.model)
+        if workspace_root is None:
+            assert task
+            workspace_root = set_workspace(task, model)
         builder.build_task_config(
-            task_type=args.task,
-            model_type=args.model,
-            train_type=args.train_type.lower(),
-            workspace_path=Path(args.workspace_root),
+            task_type=task,
+            model_type=model,
+            train_type=train_type.lower(),
+            workspace_path=Path(workspace_root),
             otx_root=otx_root,
+            template=template,
         )
         config_manager.write_data_with_cfg(
-            workspace_dir=args.workspace_root,
-            train_data_roots=args.train_data_roots,
-            val_data_roots=args.val_data_roots,
+            workspace_dir=workspace_root,
+            train_data_roots=train_data_roots,
+            val_data_roots=val_data_roots,
         )
 
     # Build Backbone related
-    if args.backbone:
+    if backbone:
```

Comment:
How about tying it into another function from here on out? (line 121- 133)

Reply:
I agree, Maybe we need to refactor the `builder` and `build` and `train`. How about doing this another PR?